src/slack/slack_client.bal

```diff
@@ -0,0 +1,390 @@
+
+    private http:Client conversationClient;
+    private map<string> idMap;
+
+    function __init(http:Client slackClient, map<string> channelIdMap) {
+        self.conversationClient = slackClient;  
+        self.idMap = channelIdMap;
+    }
+
+    # The `ConversationClient.createConversation()` function can be used to create a conversation.
+    #
+    # + name - Name of the conversation(Channel) to be created
+    # + isPrivate - true if the a private channel, false if a public channel
+    # + return - An error if it is a failure or the Channel record if it is a success
+    public remote function createConversation(string name, boolean isPrivate = false) returns @tainted Channel|error {
+        string url = CREATE_CONVERSATION_PATH + name + IS_PRIVATE_CONVERSATION + isPrivate.toString();
+        return createChannel(self.conversationClient, url);
+    }
+
+    # The `ConversationClient.archiveConversation()` function can be used to archive a conversation.
+    #
+    # + channelName - Name of the conversation to archive
+    # + return - An `error` if it is a failure or `nil` if it is a success
+    public remote function archiveConversation(string channelName) returns @tainted error|() {
+        var resolvedChannelId = check self.resolveChannelId(self.idMap, channelName);
+        return archiveConversation(self.conversationClient, <@untainted> resolvedChannelId);
+    }
+
+    # The `ConversationClient.unArchiveConversation()` function can be used to unarchive a conversation.
+    #
+    # + channelName - Name of the conversation to unarchive
+    # + return - An `error` if it is a failure or `nil` if it is a success
+    public remote function unArchiveConversation(string channelName) returns @tainted error|() {
+        var resolvedChannelId = check self.resolveChannelId(self.idMap, channelName);
+        return unArchiveConversation(self.conversationClient, <@untainted> resolvedChannelId);
+    }
+
+    # The `ConversationClient.renameConversation()` function can be used to rename a conversation.
+    #
+    # + channelName - Name of the conversation/channel
+    # + newName - 	New name for conversation.
+    # + return - An `error` if it is a failure or `Channel` record if it is a success
+    public remote function renameConversation(string channelName, string newName) returns @tainted Channel|error {
+        var resolvedChannelId = check self.resolveChannelId(self.idMap, channelName);
+        return renameConversation(self.conversationClient, <@untainted> resolvedChannelId, newName);
+    }
+
+    # The `ConversationClient.listConversations()` function can be used to list all channels in a slack team.
+    #
+    # + return - An `error` if it is a failure or `Conversations` record if it is a success
+    public remote function listConversations() returns @tainted Conversations|error {
+        http:Client convClient = self.conversationClient;
+        var response = convClient->get(LIST_CONVERSATIONS_PATH);
```

Comment:
Shall we use specific union instead of var here, since this is a union of two types, as we agreed earlier? 

Check other places as well.
```suggestion
        http:Response|error response = convClient->get(LIST_CONVERSATIONS_PATH);
```